file/builder.go

```diff
@@ -381,6 +403,37 @@ func (b *stateBuilder) ingestACLGroups(creds []kong.ACLGroup) error {
 	return nil
 }
 
+func (b *stateBuilder) ingestMTLSAuths(creds []kong.MTLSAuth) error {
+	for _, cred := range creds {
+		cred := cred
+		if utils.Empty(cred.ID) {
+			// normally, we'd want to look up existing resources in this case
+			// however, this is impossible here: mtls-auth simply has no unique fields other than ID,
+			// so we just give up and complain about it
+			var consumerFriendlyName *string
+			if !utils.Empty(cred.Consumer.Username) {
+				consumerFriendlyName = cred.Consumer.Username
+			} else if !utils.Empty(cred.Consumer.CustomID) {
+				consumerFriendlyName = cred.Consumer.CustomID
+			} else {
+				// this shouldn't happen, but apparently can. failsafe in case:
+				consumerFriendlyName = cred.Consumer.ID
+			}
+			return errors.Errorf("mtls-auth for Consumer '%s' with SubjectName '%s' lacks ID",
```

Comment:
Please remove this and instead add the following patch to take care of this using json schema validation:

```patch
diff --git a/file/codegen/main.go b/file/codegen/main.go
index 0c9f588..4d1d47e 100644
--- a/file/codegen/main.go
+++ b/file/codegen/main.go
@@ -96,6 +96,7 @@ func main() {
        schema.Definitions["KeyAuth"].Required = []string{"key"}
        schema.Definitions["Oauth2Credential"].Required = []string{"name",
                "client_id", "redirect_uris", "client_secret"}
+       schema.Definitions["MTLSAuth"].Required = []string{"id", "subject_name"}
 
        // Foreign references
        stringType := &jsonschema.Type{Type: "string"}

```

This is not as user friendly but this is more maintainable and follows the pattern we follow in the rest of the codebase.

Reply:
Make sure to run `go generate ./...` once you add the above patch.